Project/Source Files/FilesProvider/SearchFilesRequest.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  SearchFilesRequest.swift
+//  NetguruCollegeApp
+//
+
+import Foundation
+
+final class SearchFilesRequest: APIRequest {
+    var path: String = ""
+    var method: APIMethod = .get
+    var parameters: [String: String]?
+
+    init(withParameters parameters: [String: String], ownerName owner: String, repoName repo: String) {
```

Comment:
please explicitly name parameters and create parameters dictionary under the hood, like here: